doc/refman/RefMan-ext.tex

```diff
@@ -1745,6 +1745,34 @@ \subsection{Explicit displaying of implicit arguments for pretty-printing
 
 \SeeAlso {\tt Set Printing All} in Section~\ref{SetPrintingAll}.
 
+\subsection{Printing details of records and inductives}
+\optindex{Printing Records Desugared}
+
+By default, only the high-level details of records and inductives are
```

Comment:
What is a "detail"?
Maybe an example could be added, like
```coq
Record R := { n : nat }.
Set ...
Check ...
Unset ...
Check ...
```

Reply:
Oh dear, it's the printing of record types that are affected, not instances of those types.

There's an example in the manual:
```
Record Rat : Set := mkRat
  {sign : bool;
   top : nat;
   bottom : nat;
   Rat_bottom_cond : 0 <> bottom;
   Rat_irred_cond :
    forall x y z:nat, (x * y) = top /\ (x * z) = bottom -> x = 1}.
```
For `Print Rat`, the "desugared" version is:
```
Variant Rat : Set :=  mkRat : bool -> forall top bottom : nat, 0 <> bottom -> (forall x y z : nat, x * y = top /\ x * z = bottom -> x = 1) -> Rat
```
while the "sugared" version is
```
Record Rat : Set := mkRat { sign : bool;  top : nat;  bottom : nat;  Rat_bottom_cond : 0 <> bottom;  Rat_irred_cond : forall x y z : nat, x * y = top /\ x * z = bottom -> x = 1 }
```
So the notion of sugar is odd, or perhaps backwards.

I showed an example on Gitter yesterday using an Inductive, but the difference there was apparently due to other effects of `Printing All`, not the piece of code guarded in this PR by `Printing Records Sugared`.

OK: a better name for the option could be `Printing Record Types As Variants`.

Reply:
What's the purpose of this option? TBH I would just drop the option and always print records as records.

Reply:
I'm in favor of not adding options, which is just more cognitive burden for users. I'll make that change.

Reply:
Option removed.